homeassistant/components/denonavr/__init__.py

```diff
@@ -66,6 +70,10 @@ async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
         config_entry, PLATFORMS
     )
 
+    if config_entry.options.get(CONF_USE_TELNET, DEFAULT_USE_TELNET):
+        receiver: DenonAVR = hass.data[DOMAIN][config_entry.entry_id][CONF_RECEIVER]
+        await receiver.async_telnet_disconnect()
```

Comment:
Does it need to disconnect when the stop event is fired as well?

Reply:
Very likely, but not familiar with capturing the stop event, have an example?

Reply:
Found out how to do this but stuck with a weird mypy error.

Reply:
Resolved in https://github.com/home-assistant/core/pull/85980/commits/1871f2de7d133af849cab44a54070af76dcadf1c